Added support for fetching basic document information for the admin service

simonstenstrom opened the pull request:
- Added two endpoints /documents/count and /documents/list 
- The first returns the number of documents matching a query (by adding for example ?q={fetched:true})
- The second returns the documents matching a query (by adding for example ?q={fetched:true}). You can also specify the number of documents to return (by adding for example &limit=10) and skip a number of documents for pagination (by adding for example &skip=10)
- I no query is specified, all documents are considered matches.

Commit:
Changed the default oldMaxSize to 200 and oldMaxCount to 2000. So now the capped oldDocuments folder will be the same size when created by core as when created by the inserter.

Commit:
added project name so that the jar-with-dependencies would get a nice name. Also upgraded the hydra-api dependency from 0.2.0 to 0.3.0-SNAPSHOT

simonstenstrom committed:
Added methods for converting unspecified DatabaseDocument and DatabaseQuery into typed versions

simonstenstrom committed:
Added methods for converting unspecified DatabaseDocument and DatabaseQuery into typed versions

simonstenstrom committed:
Added content not equals to the Query interface and implemented it for LocalDocument. Added getters for fields in LocalDocument

simonstenstrom committed:
Added content not equals to the Query interface and implemented it for LocalDocument. Added getters for fields in LocalDocument

simonstenstrom committed:
Added methods for getting the number of documents matching a query, and getting a list of documents skipping a few documents

simonstenstrom committed:
Added methods for getting the number of documents matching a query, and getting a list of documents skipping a few documents

simonstenstrom committed:
Changing test so that it fails when no documents are found rather than thorwing nullpointer

simonstenstrom committed:
Changing test so that it fails when no documents are found rather than thorwing nullpointer

simonstenstrom committed:
Added helper methods for require fetched and not fetched by stage. Changed how a query is parsed (using its own method rather than relying on LocalDocument). Added fromJson method

simonstenstrom committed:
Added helper methods for require fetched and not fetched by stage. Changed how a query is parsed (using its own method rather than relying on LocalDocument). Added fromJson method

simonstenstrom committed:
Added implementation of interface DatabaseQuery to be used for querying and DatabaseType

simonstenstrom committed:
Added implementation of interface DatabaseQuery to be used for querying and DatabaseType

simonstenstrom committed:
Added a document service that queries for document related service requests

simonstenstrom committed:
Added a document service that queries for document related service requests

simonstenstrom committed:
Added configuration and endpoints for /documents/count and /documents/list

simonstenstrom committed:
Added configuration and endpoints for /documents/count and /documents/list

simonstenstrom committed:
Added deafult query for list

simonstenstrom committed:
Added deafult query for list

simonstenstrom committed:
Added getter for the content map so that it is shown in the output

simonstenstrom committed:
Added getter for the content map so that it is shown in the output

Commit:
changed the description of the tika stage from solr-out to Tika Text Extraction Stage

jwestberg commented on api/src/main/java/com/findwise/hydra/local/LocalQuery.java:
```diff
@@ -115,4 +121,41 @@ public void fromJson(String json) throws JsonException{
+		return notEquals;
+	}
+
+	public void setNotEquals(Map<String, Object> notEquals) {
```
Shouldn't this be modified by reference instead? Why would you ever need to setNotEquals()? getNotEquals should never return null, so it should really just be modified (changed by reference).

simonstenstrom replied:
Agreed. Changes pushed

jwestberg commented on database-impl/mongodb/src/main/java/com/findwise/hydra/mongodb/MongoConnector.java:
```diff
@@ -222,4 +224,23 @@ public MongoStatusIO getStatusWriter() {
 	}
+
+	@Override
+	public DatabaseDocument<MongoType> convert(DatabaseDocument<?> document) {
```
We could probably merge this convert(DatabaseDocument<?> and the convert(LocalDocument) to a simple convert(Document), no? Since Document extends the JsonDeserializer interface, this shouldn't be a big problem.

Seems easier to do that than to commit a bunch of Database-API changes and implement those all over :)

jwestberg commented on database/src/main/java/com/findwise/hydra/DatabaseConnector.java:
```diff
@@ -36,9 +36,12 @@
 	DocumentWriter<T> getDocumentWriter();
 	
 	DatabaseQuery<T> convert(LocalQuery query);
+	DatabaseQuery<T> convert(DatabaseQuery<?> query);
```
-38
-39
- DatabaseQuery<T> convert(Query query); ? :)

jwestberg replied:
-38
-39
- DatabaseQuery<T> convert(Query query); ? :)

laserval committed:
Merge pull request #98 from rgraneru/master

pom fixing in tikastage + CmdLineInsterter creates oldDocuments collection of same size as Core does

simonstenstrom committed:
Removed setters for fields since they are already instanciated. Moved getters for easier reading. Added getContentNotEquals returning the same map as getNotEquals to match getContentEquals

simonstenstrom committed:
Removed setters for fields since they are already instanciated. Moved getters for easier reading. Added getContentNotEquals returning the same map as getNotEquals to match getContentEquals

simonstenstrom committed:
Updated convert(DatabaseDocument) and convert(LocalDocument) into convert(Document)

simonstenstrom committed:
Updated convert(DatabaseDocument) and convert(LocalDocument) into convert(Document)

simonstenstrom committed:
Updated convert(DatabaseDocument) and convert(LocalDocument) into convert(Document) in interface

simonstenstrom committed:
Updated convert(DatabaseDocument) and convert(LocalDocument) into convert(Document) in interface

simonstenstrom commented:
Fixes committed

simonstenstrom committed:
Alos, fixed convert methods for the Momorydb implemenation

simonstenstrom committed:
Alos, fixed convert methods for the Momorydb implemenation

laserval committed:
Merge pull request #104 from jwestberg/java_version

JVM selection

simonstenstrom commented:
Rebased done

simonstenstrom committed:
Merge branch 'DocOverview' of https://github.com/simonstenstrom/Hydra into DocOverview

simonstenstrom committed:
Merged conflicts

simonstenstrom committed:
Merge branch 'DocOverview' of https://github.com/simonstenstrom/Hydra into DocOverview

simonstenstrom commented:
Added support for monitoring stages and adding/updating stage configuration. Also updated readme file with documentation on how the admin service should be used for the new features

simonstenstrom commented:
Any progress with this merge?

rgraneru commented:
Sorry about the delay. I plan to test it tomorrow.
On Nov 6, 2012 3:17 PM, "simonstenstrom" notifications@github.com wrote:

> Any progress with this merge?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/Findwise/Hydra/pull/100#issuecomment-10111916.
